### PR TITLE
Fix: plugins don't update their internal representation of config files on file change

### PR DESCRIPTION
**Pull request by Ruadhri17**
## Proposed changes
Fix the problem of not updating the internal representation of config files. This applies to all plugins that use `FsFileWatcher`: 
-  `tedge-log-plugin`
-  `c8y-log-plugin`
- `c8y-configuration-plugin`

Additionally this PR fixes unsafe `unwrap` while sending mqtt message inside the  `c8y-configuration-plugin` (that could lead to rust panic if there was any problem with MQTT actor)

The integration tests were also extended to cover these scenarios to ensure it functions as expected (taken from https://github.com/thin-edge/thin-edge.io/pull/2260):

* add library function to check service status of child devices and switch to new tedge interface
* use recent timestamp instead of 1970 as this is more aligned to how the UI behaves
* fix timestamp bug where the tests were using a fixed timezone which meant that the local time was being erroneously cast to UTC leading to unexpected dateFrom/dateTo values
* use tedge config to control `te` topic instead of flags
* add notes about when workarounds in tests can be removed

## Types of changes

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Improvement (general improvements like code refactoring that doesn't explicitly fix a bug or add any new functionality)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Paste Link to the issue
* #2246
<br/>

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTOR-LICENSE-AGREEMENT.md) (in all commits with git commit -s)
- [x] I ran `cargo fmt` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [x] I used `cargo clippy` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by reubenmiller** [comment]
@Ruadhri17 I merged the changes from https://github.com/thin-edge/thin-edge.io/pull/2260 as the integration tests were covering this aspect, so we can get confirmation that everything works as expected before merging (I also rebased ontop of the latest from main)

File: tests/RobotFramework/tests/cumulocity/log/log_operation_child.robot
Comment (albinsuresh) [inline tests/RobotFramework/tests/cumulocity/log/log_operation_child.robot:30]: That needs to be set explicitly? I thought that was the default.

File: tests/RobotFramework/tests/cumulocity/log/log_operation_child.robot
Comment (albinsuresh) [inline tests/RobotFramework/tests/cumulocity/log/log_operation_child.robot:34]: That looks weird. If the intent was just to pickup the new configs, couldn't we have just restarted the already installed plugin service? Why reinstall?

**Comment by albinsuresh** [comment]
The fix looks good. Happy to approve once my queries are addressed.

File: tests/RobotFramework/tests/cumulocity/log/log_operation_child.robot
Comment (reubenmiller) [reply tests/RobotFramework/tests/cumulocity/log/log_operation_child.robot:34]: It is not being reinstalled, it is being installed for the first time as it a child device, not a main device.

File: tests/RobotFramework/tests/cumulocity/log/log_operation_child.robot
Comment (reubenmiller) [reply tests/RobotFramework/tests/cumulocity/log/log_operation_child.robot:30]: I wanted to be explicit (that is generally my preference)

File: tests/RobotFramework/tests/cumulocity/log/log_operation_child.robot
Comment (albinsuresh) [reply tests/RobotFramework/tests/cumulocity/log/log_operation_child.robot:34]: Aah, just noticed the `_` in the first `sudo dpkg -i packages/tedge_*.deb` setup command above. And this plugin starts with `tedge-`.

**Review by albinsuresh — APPROVED** [review]

**Pull request merged**
